### ISSUE TITLE
Mapping Dictionaries with nested dictionaries

### ISSUE THREAD

**Issue by kwaclaw** [agileobjects/AgileMapper#8]
Given these two classes:

```
        class MySettings
        {
            public double Field1 { get; set; }
            public decimal Field2 { get; set; }
        }

        class FormatOptions
        {
            public int CharactersPerInch { get; set; }
            public long LinesPerPage { get; set; }
            public string Dummy { get; set; }
            public MySettings Settings { get; set; }
        }

```

and this input: 
```
            var exportFormatOptions = new Dictionary<string, object>
            {
                { "CharactersPerInch", 16 },
                { "LinesPerPage", (long)0 },
                { "Dummy", "Test Dummy" },
                { "Settings", new Dictionary<string, object>
                    {
                        { "Field1", (double)22.3 },
                        { "Field2", (decimal)2345 }
                    }
                }
            };

            var options = Mapper.Map(exportFormatOptions).ToANew<FormatOptions>();
```

The nested Settings property is not mapped and I cannot find a way to tell the mapper to treat the Settings entry in the outer dictionary as a Dictionary<string, object) (its runtime type).

**Comment by SteveWilkes** [comment]
This is a bug which I'll fix with a new release this week - the mapper isn't taking the fact that it's a nested dictionary into account and is looking for the keys "Settings.Field1" and "Settings.Field2" instead of "Field1" and "Field2". You should be able to around it in the meantime using:

``` C#
var exportFormatOptions = new Dictionary<string, object>
{
    { "CharactersPerInch", 16 },
    { "LinesPerPage", 0 },
    { "Dummy", "Test Dummy" },
    { "Settings", new Dictionary<string, object>
        {
            { "Settings.Field1", 22.3 },
            { "Settings.Field2", 2345 }
        }
    }
};
```

Incidentally you don't need to cast the numeric values - the Mapper will parse them for you :)

**Comment by SteveWilkes** [comment]
Fixed in v0.10.1 - thanks for the bug report! :)

If you used the workaround, you'll have to change the keys back to how they were originally.
